### PR TITLE
 pkg/metrics/firewall.go add firewall status

### PR DESCRIPTION
**Pull request by tobiasgiese**
Adds Firewall metrics for FWaaS v1

### CODE REVIEW

**Review by sbueringer — APPROVED** [review]

File: pkg/metrics/firewall.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		append(firewallLabels, "status"),
+	)
+
+	prometheus.MustRegister(firewallAdminStateUp)
+	prometheus.MustRegister(firewallStatus)
+}
+
+// PublishFirewallMetrics makes the list request to the firewall api and
+// passes the result to a publish function.
+func PublishFirewallMetrics(client *gophercloud.ServiceClient, tenantID string) error {
```
Comment (chrischdi) [inline pkg/metrics/firewall.go:45]:
```suggestion
func PublishFWaaSV1Metrics(client *gophercloud.ServiceClient, tenantID string) error {
```

We may later want to implement FWaaSV2 metrics too

File: pkg/metrics/firewall.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	"k8s.io/klog"
+)
+
+var (
+	firewallAdminStateUp *prometheus.GaugeVec
+	firewallStatus       *prometheus.GaugeVec
+
+	// according to the nl_constants from https://github.com/openstack/neutron-fwaas/blob/stable/ocata/neutron_fwaas/services/firewall/fwaas_plugin.py
+	// we will get the following firewall states
+	firewallStates = []string{"ACTIVE", "DOWN", "ERROR", "INACTIVE", "PENDING_CREATE", "PENDING_UPDATE", "PENDING_DELETE"}
```
Comment (chrischdi) [inline pkg/metrics/firewall.go:18]:
```suggestion
	firewallV1States = []string{"ACTIVE", "DOWN", "ERROR", "INACTIVE", "PENDING_CREATE", "PENDING_UPDATE", "PENDING_DELETE"}
```

File: pkg/metrics/firewall.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+	firewallLabels = []string{"id", "name", "description", "policyID", "projectID"}
+)
+
+func registerFirewallMetrics() {
+	firewallAdminStateUp = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("firewall_admin_state_up"),
```
Comment (chrischdi) [inline pkg/metrics/firewall.go:26]:
```suggestion
			Name: generateName("firewall_v1_admin_state_up"),
```
also for the others pls

File: pkg/metrics/firewall.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+	// according to the nl_constants from https://github.com/openstack/neutron-fwaas/blob/stable/ocata/neutron_fwaas/services/firewall/fwaas_plugin.py
+	// we will get the following firewall states
+	firewallStates = []string{"ACTIVE", "DOWN", "ERROR", "INACTIVE", "PENDING_CREATE", "PENDING_UPDATE", "PENDING_DELETE"}
+
+	firewallLabels = []string{"id", "name", "description", "policyID", "projectID"}
+)
+
+func registerFirewallMetrics() {
+	firewallAdminStateUp = prometheus.NewGaugeVec(
```
Comment (chrischdi) [inline pkg/metrics/firewall.go:24]:
```suggestion
	firewallV1AdminStateUp = prometheus.NewGaugeVec(
```
Also for the others pls

File: pkg/metrics/firewall.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
```
Comment (chrischdi) [inline pkg/metrics/firewall.go:1]: maybe it does make sense to rename the file to `fwaasv1.go` ?

File: pkg/metrics/fwaasv1.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	}
+	firewallsList, err := firewalls.ExtractFirewalls(pages)
+	if err != nil {
+		// only warn, maybe the next publish will work.
+		klog.Warningf("Unable to extract firewalls: %v", err)
+		return err
+	}
+
+	// second step: reset the old metric
+	firewallV1Status.Reset()
```
Comment (chrischdi) [inline pkg/metrics/fwaasv1.go:62]:
```suggestion
	firewallV1Status.Reset()
	firewallV1AdminStateUp.Reset()
```

maybe I also missed this in loadbalancers.go?

Comment (tobiasgiese) [reply pkg/metrics/fwaasv1.go:62]: Yep, and I have also asked myself why

**Pull request merged**